### PR TITLE
Add minimum logging to run train_from_scratch()

### PR DESCRIPTION
**Pull request by zurk**
Logging should be improved in a lot of places related to typos.
I add just a basic logging needed to understand what is going on while running ` train_from_scratch()`

### CODE REVIEW

File: lookout/style/typos/corrector.py [hunk 1]
```diff
@@ -75,6 +80,7 @@ def initialize_ranker(self, boost_params: Optional[dict] = None,
         """
         boost_params = boost_params or self.DEFAULT_BOOST_PARAM
         self.ranker.construct(boost_params, train_rounds, early_stopping)
+        self._log.debug("Ranker %s is initialized.", repr(self.ranker))
```
Comment (vmarkovtsev) [inline lookout/style/typos/corrector.py]: If `repr` includes the class name, we should remove redundant "Ranker" from the logs.

File: lookout/style/typos/corrector.py [hunk 2]
```diff
@@ -97,6 +103,7 @@ def initialize_generator(self, vocabulary_file: str, frequencies_file: str,
         """
         self.generator.construct(vocabulary_file, frequencies_file, embeddings_file,
                                  neighbors_number, edit_candidates, max_distance, radius)
+        self._log.debug("Generator %s is initialized.", repr(self.generator))
```
Comment (vmarkovtsev) [inline lookout/style/typos/corrector.py]: same here

File: lookout/style/typos/corrector.py [hunk 3]
```diff
@@ -120,11 +127,14 @@ def train(self, data: pandas.DataFrame, candidates:  Optional[str] = None,
         :param start_pool_size: Length of data, starting from which multiprocessing is desired.
         :param chunksize: Max size of a chunk for one process during multiprocessing.
         """
+        self._log.info("Training on data with shape %s.", data.shape)
         if candidates is None:
+            self._log.info("Candidates are not provided and will be generated.")
```
Comment (vmarkovtsev) [inline lookout/style/typos/corrector.py]:
```suggestion
            self._log.info("Candidates were not provided and will be generated.")
```

File: lookout/style/typos/corrector.py [hunk 4]
```diff
@@ -120,11 +127,14 @@ def train(self, data: pandas.DataFrame, candidates:  Optional[str] = None,
         :param start_pool_size: Length of data, starting from which multiprocessing is desired.
         :param chunksize: Max size of a chunk for one process during multiprocessing.
         """
+        self._log.info("Training on data with shape %s.", data.shape)
```
Comment (vmarkovtsev) [inline lookout/style/typos/corrector.py]:
```suggestion
        self._log.info("Train input: %s of %s", data.shape, data.dtype)
```

Comment (vmarkovtsev) [reply lookout/style/typos/corrector.py]: (it is better to make logs shorter and don't write long NL sentences - they make it longer to understand the logs)

File: lookout/style/typos/corrector.py [hunk 5]
```diff
@@ -120,11 +127,14 @@ def train(self, data: pandas.DataFrame, candidates:  Optional[str] = None,
...
         if candidates is None:
+            self._log.info("Candidates are not provided and will be generated.")
             candidates = self.generator.generate_candidates(
                 data, self.processes_number, start_pool_size, chunksize, save_candidates_file)
         else:
             candidates = pandas.read_csv(candidates, index_col=0, keep_default_na=False)
+            self._log.info("Candidates are loaded from %s.", candidates)
```
Comment (vmarkovtsev) [inline lookout/style/typos/corrector.py]:
```suggestion
            self._log.info("Loaded candidates from %s", candidates)
```

File: lookout/style/typos/preparation.py [hunk 6]
```diff
@@ -109,37 +113,51 @@ def prepare_data(config: Optional[Mapping[str, Any]] = None) -> pandas.DataFrame
     if raw_data_path is None or not os.path.exists(raw_data_path):
         raw_data_path = os.path.join(config["data_dir"],
                                      config["raw_data_filename"])
+        log.info("Raw dataset is not found. Downloading to %s from %s",
```
Comment (vmarkovtsev) [inline lookout/style/typos/preparation.py]:
```suggestion
        log.warning("Raw dataset was not found. Downloading to %s from %s",
```

Comment (vmarkovtsev) [reply lookout/style/typos/preparation.py]: Order: first "from", second "to" 😄

File: lookout/style/typos/preparation.py [hunk 7]
```diff
@@ -109,37 +113,51 @@ def prepare_data(config: Optional[Mapping[str, Any]] = None) -> pandas.DataFrame
...
                                      apply_function=lambda x: x.split())
+    log.debug("Expanded data shape %s", flat_data.shape)
 
-    # Collect statistics for tokens
+    log.info("Collect statistics for tokens")
     stats = flat_data[[Columns.Frequency, Columns.Token]].groupby([Columns.Token]).sum()
     stats = stats.sort_values(by=Columns.Frequency, ascending=False)
 
-    # Derive new vocabulary for future use
+    log.info("Derive new vocabulary")
```
Comment (vmarkovtsev) [inline lookout/style/typos/preparation.py]:
```suggestion
    log.info("Derive the new vocabulary")
```

File: lookout/style/typos/preparation.py [hunk 8]
```diff
@@ -109,37 +113,51 @@ def prepare_data(config: Optional[Mapping[str, Any]] = None) -> pandas.DataFrame
...
     frequencies_tokens = set(stats.index[:(config["frequencies_size"] or len(stats))])
+    log.info("Tokens with frequencies data size: %d", len(frequencies_tokens))
     vocabulary_tokens = set(stats.index[:config["vocabulary_size"]])
-    print_frequencies(vocabulary_tokens, stats, os.path.join(
-        config["data_dir"], config["vocabulary_filename"]))
-    print_frequencies(frequencies_tokens, stats, os.path.join(
-        config["data_dir"], config["frequencies_filename"]))
+    log.info("vocabulary size: %d", len(vocabulary_tokens))
```
Comment (vmarkovtsev) [inline lookout/style/typos/preparation.py]: The logs should all start with a capital letter or with a small letter

File: lookout/style/typos/preparation.py [hunk 9]
```diff
@@ -109,37 +113,51 @@ def prepare_data(config: Optional[Mapping[str, Any]] = None) -> pandas.DataFrame
...
+    vocabulary_tokens_filepath = os.path.join(config["data_dir"], config["vocabulary_filename"])
+    print_frequencies(vocabulary_tokens, stats, vocabulary_tokens_filepath)
+    log.info("Vocabulary saved to %s", vocabulary_tokens_filepath)
+    frequencies_tokens_filepath = os.path.join(config["data_dir"], config["frequencies_filename"])
+    print_frequencies(frequencies_tokens, stats, frequencies_tokens_filepath)
+    log.info("Tokens with frequencies data saved to %s", frequencies_tokens_filepath)
```
Comment (vmarkovtsev) [inline lookout/style/typos/preparation.py]:
```suggestion
    log.info("Tokens with frequencies data are saved to %s", frequencies_tokens_filepath)
```

File: lookout/style/typos/ranking.py [hunk 10]
```diff
@@ -69,6 +73,10 @@ def fit(self, identifiers: pandas.Series, candidates: pandas.DataFrame,
         :param features: Matrix of features for candidates.
         :param val_part: Part of data used for validation.
         """
+        self._log.info("Fitting is started.")
```
Comment (vmarkovtsev) [inline lookout/style/typos/ranking.py]:
```suggestion
        self._log.info("Fitting has started.")
```

File: lookout/style/typos/corrector.py (hunk 4 above)
Comment (zurk) [reply lookout/style/typos/corrector.py]: dtype is always the same so I'd rather use
```python
self._log.info("Train input shape: %s", data.shape)
```

Comment (vmarkovtsev) [reply lookout/style/typos/corrector.py]: wdym the same? I will call this function on an array of strings and watch it burn in fire.

After all, the purpose of this line is to be a replacement for `repr`, but without the array contents. Let's be consistent.

**Comment by zurk** [comment]
Thanks, @vmarkovtsev, fixed!

File: lookout/style/typos/preparation.py (hunk 8 above)
Comment (vmarkovtsev) [reply lookout/style/typos/preparation.py]: @zurk Missed this ^

Comment (zurk) [reply lookout/style/typos/preparation.py]: All logs start with small letter :)

Comment (vmarkovtsev) [reply lookout/style/typos/preparation.py]: Oh 😄

**Pull request merged**